server/tools/client_credentials_admin.py

```diff
@@ -0,0 +1,179 @@
+# Copyright (C) 2022 Canonical
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+"""
+This script administers client credentials in a MongoDB database. Database login
+information must be specified with environment variables. The script prompts user
+to create a new entry, or edit/delete an existing entry. The user will then be prompted to enter new max_priority and allowed_queues data.
+
+"""
+
+import sys
+import json
+import bcrypt
+import os
+import urllib
+
+from pymongo import MongoClient, ReturnDocument
+
+
+def setup_database():
+    mongo_user = os.environ.get("MONGODB_USERNAME")
+    mongo_pass = os.environ.get("MONGODB_PASSWORD")
+    if mongo_pass:
+        # password might contain special chars
+        mongo_pass = urllib.parse.quote_plus(mongo_pass)
+
+    mongo_db = os.environ.get("MONGODB_DATABASE")
+    mongo_host = os.environ.get("MONGODB_HOST")
+    mongo_port = os.environ.get("MONGODB_PORT", "27017")
+    mongo_uri = os.environ.get("MONGODB_URI")
+    mongo_auth = os.environ.get("MONGODB_AUTH_SOURCE", "admin")
+
+    if not mongo_uri:
+        if not (mongo_host and mongo_db):
+            raise SystemExit("No MongoDB URI configured!")
+        mongo_creds = (
+            f"{mongo_user}:{mongo_pass}@" if mongo_user and mongo_pass else ""
+        )
+        mongo_uri = (
+            f"mongodb://{mongo_creds}{mongo_host}:{mongo_port}/{mongo_db}"
+            f"?authSource={mongo_auth}"
+        )
+
+    client = MongoClient(
+        host=mongo_uri,
+        username=mongo_user,
+        password=mongo_pass,
+    )
+
+    return client[mongo_db]
+
+
+def handle_max_priority_input() -> dict:
+    """Gets a user inputted max_priority dict"""
+    max_priority = {}
+    while True:
+        max_priority_input = input(
+            "Do you want to specify a max priority for one or more queues? (y/n): "
```

Comment:
I feel like the wording of this is unclear... maybe something more like "Specify a queue to set max priority on?" Unless you can think of something better?